ff-fft/src/domain/mod.rs

```diff
@@ -0,0 +1,219 @@
+//! This module contains an `EvaluationDomain` abstraction for
+//! performing various kinds of polynomial arithmetic on top of
+//! the scalar field.
+//!
+//! In pairing-based SNARKs like GM17, we need to calculate
+//! a quotient polynomial over a target polynomial with roots
+//! at distinct points associated with each constraint of the
+//! constraint system. In order to be efficient, we try to choose these
+//! roots to be the powers of a 2^n root of unity in the field.
+//! This allows us to perform polynomial operations in O(n)
+//! by performing an O(n log n) FFT over such a domain.
+//!
+//! If the 2-adicity of the field is too small, but a small subgroup
+//! over the field is defined, we can try to build a mixed-radix evaluation
+//! domain.
+
+use crate::Vec;
+use algebra_core::FftField;
+use core::{fmt, hash};
+use rand::Rng;
+#[cfg(feature = "parallel")]
+use rayon::prelude::*;
+
+pub mod general;
+pub mod mixed_radix;
+pub mod radix2;
+pub(crate) mod utils;
+
+pub use general::GeneralEvaluationDomain;
+pub use mixed_radix::MixedRadixEvaluationDomain;
+pub use radix2::Radix2EvaluationDomain;
+
+/// Defines a domain over which finite field (I)FFTs can be performed. Works
+/// only for fields that have a large multiplicative subgroup of size that is
+/// a power-of-2.
```

Comment:
This is no longer accurate, right?

Reply:
Completely true. Thanks for taking care of the comments. :)